include/zephyr/llext/buf_loader.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_LLEXT_BUF_LOADER_H
+#define ZEPHYR_LLEXT_BUF_LOADER_H
+
+#include <zephyr/llext/loader.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief LLEXT buffer loader
+ * @defgroup llext_buf_loader Linkable loadable extensions buffer loader
+ * @ingroup llext
+ * @{
+ */
+
+/**
+ * @brief A llext loader context for an ELF in a buffer
+ */
+struct llext_buf_loader {
+	struct llext_loader loader;
+	const uint8_t *buf;
+	size_t len;
+	size_t pos;
```

Comment:
missing docstrings

Reply:
Everything but the loader member is meant to be private, and has been wrapped in ```@cond ignore```